### PR TITLE
feat: apply new deisgn on project tree to solve Critical create actions inaccessible issue on Project Tree

### LINKED ISSUE #5681: Nav overlaps with three dot menus - Critical create actions inaccessible
image
This is making critical action in inaccessible

Suggested fix - Increase padding from right side.

### PR DESCRIPTION
**Pull request by liweitian**
## Description
apply new design on project tree without the addTrigger and addDialog icons
https://www.figma.com/file/deFhMPBnRWxTYiYDYqYxrc/Composer-Vivek?node-id=10809%3A0

and

solve a bug on resizing the project window. 
It does not re-render when resizing because the "leftSplitWidth" is not passed into the an useCallback function as a dependency. So some very long name dialog is still hidden after expanding the window.
## Task Item
Closes #5681 

## Screenshots
split

### CODE REVIEW

File: Composer/packages/client/src/components/ProjectTree/AddDialogIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import { NeutralColors } from '@uifabric/fluent-theme';
+interface AddDialogIconProps {
+  active: boolean;
+  className?: string;
+  onClick?: () => {};
```
Comment (hatpick) [inline Composer/packages/client/src/components/ProjectTree/AddDialogIcon.tsx:10]: () => void

File: Composer/packages/client/src/components/ProjectTree/AddDialogIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import { NeutralColors } from '@uifabric/fluent-theme';
+interface AddDialogIconProps {
```
Comment (hatpick) [inline Composer/packages/client/src/components/ProjectTree/AddDialogIcon.tsx:7]: prefer type over interface

File: Composer/packages/client/src/components/ProjectTree/AddDialogIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      />
+      <path
+        d="M30.9862 23.8355H35.7533V35.7533H23.8355V23.8355H28.6026V19.0684H7.15066V23.8355H11.9178V35.7533H0V23.8355H4.76711V16.6849H16.6849V11.9178H11.9178V0H23.8355V11.9178H19.0684V16.6849H30.9862V23.8355ZM14.3013 2.38355V9.53421H21.452V2.38355H14.3013ZM9.53421 33.3698V26.2191H2.38355V33.3698H9.53421ZM33.3698 33.3698V26.2191H26.2191V33.3698H33.3698Z"
+        fill={NeutralColors.gray160}
+      />
+    </svg>
+  );
+};
+
+export default AddDialogIcon;
```
Comment (hatpick) [inline Composer/packages/client/src/components/ProjectTree/AddDialogIcon.tsx:37]: unnecessary export here, it's already exported

File: Composer/packages/client/src/components/ProjectTree/AddTriggerIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import { NeutralColors } from '@uifabric/fluent-theme';
+interface AddTriggerIconProps {
+  active: boolean;
+  className?: string;
+  id: string;
+  onClick?: () => {};
```
Comment (hatpick) [inline Composer/packages/client/src/components/ProjectTree/AddTriggerIcon.tsx:11]: () => void;

File: Composer/packages/client/src/components/ProjectTree/AddTriggerIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import { NeutralColors } from '@uifabric/fluent-theme';
+interface AddTriggerIconProps {
```
Comment (hatpick) [inline Composer/packages/client/src/components/ProjectTree/AddTriggerIcon.tsx:7]: prefer type

File: Composer/packages/client/src/components/ProjectTree/AddTriggerIcon.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      />
+      <path
+        d="M27.3335 39.9992V43.3358H33.9986V50.0008H37.3352V43.3358H44.0002V39.9992H37.3352V33.3341H33.9986V39.9992H27.3335Z"
+        fill={NeutralColors.gray160}
+      />
+    </svg>
+  );
+};
+
+export default AddTriggerIcon;
```
Comment (hatpick) [inline Composer/packages/client/src/components/ProjectTree/AddTriggerIcon.tsx:38]: unnecessary

File: Composer/packages/client/src/components/ProjectTree/ProjectTree.tsx [hunk 7]
```diff
@@ -28,11 +29,13 @@ import { useFeatureFlag } from '../../utils/hooks';
 import { LoadingSpinner } from '../LoadingSpinner';
 import TelemetryClient from '../../telemetry/TelemetryClient';
 
+//import AddDialogIcon from './AddDialogIcon';
```
Comment (hatpick) [inline Composer/packages/client/src/components/ProjectTree/ProjectTree.tsx:32]: remove

File: Composer/packages/client/src/components/ProjectTree/treeItem.tsx [hunk 8]
```diff
@@ -201,18 +209,46 @@ const itemName = (nameWidth: number) => css`
...
+  width: 24px;
+  .action-btn {
+    visibility: ${isActive ? 'visible' : 'hidden'};
+  }
+  :hover {
+    background: ${NeutralColors.gray20};
+    .action-btn {
+      visibility: visible;
+      path {
+        fill: #0078d4;
```
Comment (hatpick) [inline Composer/packages/client/src/components/ProjectTree/treeItem.tsx:230]: use accent from theme palette

File: Composer/packages/client/src/components/ProjectTree/treeItem.tsx [hunk 9]
```diff
@@ -201,18 +209,46 @@ const itemName = (nameWidth: number) => css`
...
+      }
+    }
+  }
+`;
+
 // -------------------- TreeItem -------------------- //
 
 interface ITreeItemProps {
   link: TreeLink;
+  ActionIcon?: React.FC<any>;
```
Comment (hatpick) [inline Composer/packages/client/src/components/ProjectTree/treeItem.tsx:240]: This is awkward way of doing this. Instead pass an onRenderActionIcon instead: (whatever args you want) => React.ReactNode

**Review by hatpick — CHANGES_REQUESTED** [review]

**Comment by hatpick** [comment]
As for the two svg Icons you are creating, I highly recommend looking into this:

import { createSvgIcon } from '@fluentui/react-icons';

from fabric.

Considering you have the svg code for the icons you can do this:

const myIconSvg = ` the svg code `; // svg code for the icon in string format
const MyIcon = createSvgIcon({ svg: () => myIconSvg, displayName: 'MyIcon' }); // Icon component showing the svg icon

now you can use MyIcon like <MyIcon style={ any style of type React.CSSProperties} />

**Comment by liweitian** [comment]
> As for the two svg Icons you are creating, I highly recommend looking into this:
> 
> `import { createSvgIcon } from '@fluentui/react-icons';`
> 
> from fabric.
> 
> Considering you have the svg code for the icons you can do this:
> 
> ```
> const myIconSvg = ` the svg code `; // svg code for the icon in string format
> const MyIcon = createSvgIcon({ svg: () => myIconSvg, displayName: 'MyIcon' }); // Icon component showing the svg icon
> ```
> 
> now you can use MyIcon like <MyIcon style={ any style of type React.CSSProperties} />

all comments handled

**Review by hatpick — APPROVED** [review]

**Review by boydc2014 — APPROVED** [review]

**Review by boydc2014 — APPROVED** [review]

**Pull request merged**